File: javascript/006.js
// Name:     Sum square difference
// Source:   https://projecteuler.net/problem=6


const sum_of_square = num => [...Array(num + 1).keys()].map(x => Math.pow(x, 2)).reduce((x, y) => x + y);

const square_of_sum = num => Math.pow([...Array(num + 1).keys()].reduce((x, y) => x + y), 2);

const difference = num => square_of_sum(num) - sum_of_square(num);


console.log(difference(100));

File: problems/006/006.hs
sumOfSquares :: Int -> Int
sumOfSquares x = sum $ map (^2) [1..x]

squareOfSum :: Int -> Int
squareOfSum x = (^2) $ sum [1..x]

sumSquareDiff :: Int -> Int
sumSquareDiff x = squareOfSum x - sumOfSquares x

-- > sumSquareDiff 100
